homeassistant/helpers/signal.py

```diff
@@ -17,7 +17,12 @@ def async_register_signal_handling(hass: HomeAssistant) -> None:
     if sys.platform != 'win32':
         @callback
         def async_signal_handle(exit_code):
-            """Wrap signal handling."""
+            """Wrap signal handling:
+            * queue shutdown code
+            * re-instate default handler
+            """
+            hass.loop.remove_signal_handler(signal.SIGTERM)
```

Comment:
Why would we do this ?

Reply:
Because the signal handler has initiated an orderly shutdown. Its job is now done; if we get the signal again, the user is impatient / the main loop is stuck, and standard Python signal handling should take over.